### PR TITLE
Add more logging to iis graceful shutdown

### PR DESCRIPTION
**Pull request by pakrym**
Add more logging to window finding code and crash iisexpress process if it was unable to find the window. It would allow us to get a memory dump.

### CODE REVIEW

**Comment by pakrym** [comment]
This passed with many iterations in multiple configurations.

File: src/Servers/IIS/IntegrationTesting.IIS/src/IISExpressDeployer.cs [hunk 1]
```diff
@@ -453,49 +453,91 @@ private class WindowsNativeMethods
...
         {
-            Logger.LogInformation($"Sending shutdown request to {pid}");
             var found = false;
-            WindowsNativeMethods.EnumWindows((ptr, param) => {
-                WindowsNativeMethods.GetWindowThreadProcessId(ptr, out var windowProcessId);
-                if (pid == windowProcessId)
-                {
-                    // 256 is the max length
-                    var className = new StringBuilder(256);
+            var extraLogging = false;
```
Comment (jkotalik) [inline src/Servers/IIS/IntegrationTesting.IIS/src/IISExpressDeployer.cs]: Seems odd to have this bool to avoid one log call.

Comment (pakrym) [reply src/Servers/IIS/IntegrationTesting.IIS/src/IISExpressDeployer.cs]: It's a log call that logs a lot.

File: src/Servers/IIS/IntegrationTesting.IIS/src/IISExpressDeployer.cs [hunk 2]
```diff
@@ -453,49 +453,91 @@ private class WindowsNativeMethods
...
+            try
+            {
+                Logger.LogInformation($"Trying to crash the process {pid}");
+                var process = Process.GetProcessById(pid);
+                // Calling CreateRemoteThread as 0x1 as thread function pointer should cause a crash
+                WindowsNativeMethods.CreateRemoteThread(process.Handle, IntPtr.Zero, 0, (IntPtr)1, IntPtr.Zero, 0, IntPtr.Zero);
+            }
+            catch (Exception e)
+            {
+                Logger.LogInformation("Exception while trying to crash the process", e);
```
Comment (jkotalik) [inline src/Servers/IIS/IntegrationTesting.IIS/src/IISExpressDeployer.cs]: nit: nothing logs the exception here.

**Review by jkotalik — APPROVED** [review]

**Pull request merged**